### DISCUSSION TITLE
K-9 Mail/Android 版 Thunderbird は Mozilla 日本語コミュニティで取り扱いますか？

### DISCUSSION THREAD

**Discussion by kazushi3** [mozilla-japan/translation#720] [General]
K-9 Mail を取り込んで Android 版 Thunderbird を開発しようという動きがあります。

事前情報:
[Thunderbird のブログ](https://blog.thunderbird.net/2022/06/revealed-thunderbird-on-android-plans-k9/)によると、K-9 の開発者は Thunderbird に合流して開発を行っているようです。

K-9 と Thunderbird が別のアプリとしてリリースされるのか、K-9 が Thunderbird にリネームされるのかは K-9 の開発者と Thunderbird ブログで齟齬があるのですが、多くの UI は共通化されるとみられます。翻訳プラットフォームでは K-9 Mail / Thunderbird と並列で表記されているため翻訳も 1 か所で両方行うとみられます。

本題:
K-9 Mail / Thunderbird の翻訳ツールは Pontoon ではなく、[Hosted Weblate](https://hosted.weblate.org/projects/tb-android/-/ja/) となっています (Mozilla 移行のタイミングで翻訳ツールを Transifex から Weblate に移行したため、Pontoon ではない理由・再移行の有無は K-9 Mail に[質問中](https://github.com/thundernest/k-9/issues/7275))。こちらについて Mozilla 日本語コミュニティとして取り組むか確認したいです。

なお、私が移行後に個人で K-9 Mail を翻訳したため、以前にあった表記・スタイルの揺れ・誤訳は Mozilla Japan の L10N 準拠で修正済みです。移行後追加分も今のところ追えています。もし Mozilla 日本語コミュニティとして取り扱わない場合は個人でやりますが、可能な限りコミュニティでやりたいです。

追記:
K-9 Mail の開発者より[返答](https://github.com/thundernest/k-9/issues/7275)あり。いま K-9 にいる翻訳者が Pontoon に慣れていないが、Weblate なら…とのことらしいです。Pontoon への移行予定はなしだそう。

**Comment by gurezo** [comment]
@marsf @hATrayflood @hmatrjp @dynamis 
cc @potappo 
こちら本日の OSC Tokyo 2023 fall (2023.10.21) で、相談がありました。
自分は知見がないので、DIscussion へポストするようにお話をさせて頂きました。

**Comment by marsf** [comment]
@kazushi3 さん

大事なことを 2 つ書きます。

まず、「Mozilla コミュニティ」は、Mozilla の活動に参加している個人やユーザーの集まりであって、コミュニティの代表者がいたり、組織的な取り決めがあるわけではありません。
ですから、「Mozilla 日本語コミュニティとして取り組むか」どうかを決められる人はいません。
ただし、それぞれのプロジェクトについては、コミット権限やレビュー権限を持っていたり、Mozilla と契約を交わしている人がいて、作業の決定権があるリーダーと呼ばれる人がいます。

K-9 Mail は Mozilla とは別のプロジェクトでしたから、移行期間の今はまだ Mozilla 日本語ローカライズの一部と言えるかどうか微妙なところです。
見方を変えて、K-9 のローカライズを行っている kazushi3 さんが Mozilla コミュニティに参加していれば、Mozilla コミュニティのメンバーの活動と呼べると思います。


次に、K-9 Mail は Mozilla とは別のプロジェクトとして開発されてきましたから、Mozilla コミュニティとは別にユーザーのコミュニティがあって、これまでのローカライズに貢献してきた人がいます。
ローカライズを引き取りたいのであれば、まず現コミュニティに問い合わせるのが先ではないでしょうか。

- [K-9 Mail 日本語グループ](https://web.archive.org/web/20201012102847/https://sites.google.com/site/k9mailja/) (web.archive.org)
- [日本語訳を手伝うには](https://web.archive.org/web/20201012102851/https://sites.google.com/site/k9mailja/home/translate)  (web.archive.org)
- [k9mail 日本語グループ - Google グループ](https://groups.google.com/g/k9mail_ja)

上記のサイトは Koji Arai さんという方が管理していたようですが、google サイトのほうはすでに無くなっています。
Google グループのほうでローカライズの現状がどうなっているか聞いてみてはいかがでしょうか。

**Reply by kazushi3** [reply]
ありがとうございます。
1 点目について、mozilla-japan の Github から翻訳できるものについてのみであれば理解できるのですが、SUMO や Pontoon などでの翻訳がこちらの翻訳のルールに従っていることはどのように捉えればよいのでしょうか。各コミッター・レビュアーの判断でしょうか。
このコミュニティに代表などがいないことは理解した上でディスカッションなどで緩やかに意思決定を行っているものと考えていました。

その上で 2 点目ですが、私も Google グループに連絡したり、Transifex で個別に翻訳者に連絡したりすることは考えたのですが、私が勝手に引き取りたいと言うのはまずいと思いこちらで先に確認をと思いました。
また、Transifex での日本語化では Koji Arai さんがレビュー権限を持ち、私や marsf さんを含む 14 名が参加していましたが、Weblate 移行後に翻訳した人が私以外いないようなので急ぐ必要もないと思っています。とりあえず、Koji Arai さんに今後どうなさるつもりか聞いてみようかなと思います。

追記:
移行期間ということについて、[ロードマップ](https://developer.thunderbird.net/planning/android-roadmap#k9-greater-than-thunderbird-for-android)ではリリースが Q4 2023 と表記されています。自信がないのですが、英語圏の Q4 だと 10～12 月までになるはずで、割とすぐではないでしょうか。

**Reply by hmatrjp** [reply]
mar さんのアドバイスもあり、あるべき流れに乗ったと思いますが、未回答の部分、コミュニティーのあり方を考える上で面白いので、書かせてもらいます。補足、修正、突っ込み歓迎。

> SUMO や Pontoon などでの翻訳がこちらの翻訳のルールに従っていることは

本来の疑問詞は　なぜ　で、直接の答えは　 守る必要がある人間がルールを作り、後から加わる人は変えるほどの不便がないので従っている
ではないかと思いますが、　(1)どんなルールが、という点と、(2)誰が、という2点に分けて考えた方がわかりやすいと思います。

(1)
翻訳に関するルールというと、何種類かあると思うので、いくつか挙げてみます。
翻訳を読む人の利便性を考えてのことだったり、作業に携わる人が迷ったときに参照する類のものと（私は）理解しています。

1. 訳出に使う言葉の揺れを防止するルール
翻訳の対象である MDN でもFirefox, Thunderbird 等のMozilla製品でも使われる言葉があるため、どちらからも参照されるものです。
2. 作業の重複を防ぐためのルール
 これは、翻訳ミートアップで使っている当日作業衝突防止用の掲示板などが該当するでしょう。
3. Mozilla、WebDinoとの約束事、厚意でできること、できないこと
 この Github リポジトリが使えていることとか、Slackチャネルが使えていることとか、Mozilla ロゴの使用が絡むこと、などでしょうか。

(2)
ざっくりと書いてしまうと、
何か決めておいた方がいいことを誰かが思いつく
 → 口火を切る
　→　（興味を引くトピックなら）意見がいくつか出てくる
　→　最終的にはなんらかのページ(Wiki等)に記述がまとまる、
というのがこれまでのルール制定の流れです。
なので、より楽ができるように手順を変えるとか、ルールを決めた基になった事情が変わったためルールが変わる（例：長音のーをつけるかつけないか）こともあります。

(2) の副作用として、成立した時期によって、置き場所が散逸する、という性質があります。（別issueで統合の提案が出ていましたね。）

**Reply by hmatrjp** [reply]
追記の部分について、
> 英語圏の Q4 だと 10～12 月までになるはずで、割とすぐではないでしょうか。

・Q4 が 10-12月を指しているのはおそらくそのとおりです。
・そのロードマップページの最終更新（末尾にあります）が4ヶ月前であることは注意しておいた方がいいと思います。
　書かれたのは表題にあるとおり、（北半球の）夏の時点、その後、実績の反映がされていないかも、という意味で。

「日本語コミュニティーでやりたい」と書かれた意図で、もし何か具体的に実現したい要望が何かあれば教えてください。
（例）
・人数を集めたい、
・github の mozilla-japan/translation リポジトリの機能を使いたい

そこにQ4 を締め切りと想定したものが何かあるのかな、と思った次第です。

**Comment by marsf** [comment]
11/3 付で、K-9 の Weblate に以下の通知が来ていました。
ソースコードが Thunderbird 配下になったので、これで移行が完了したとみてよいと思います。

> Hi everyone,
> 
> We've successfuly completed the move of our source code repository to https://github.com/thunderbird/thunderbird-android. Translation have been unlocked, and we greatly appreciate your contributions.
> 
> Best regards,
> 
> Wolf

日本語ローカライズのページはこちら
https://hosted.weblate.org/projects/tb-android/-/ja/

@kazushi3 
K-9 コミュニティのほうから何か応答がありましたか？

**Comment by marsf** [comment]
`SUMO や Pontoon などでの翻訳がこちらの翻訳のルールに従っていることはどのように捉えればよいのでしょうか。各コミッター・レビュアーの判断でしょうか。`

もう一つの issue で答えるべき内容のように思いますので、ここでは簡単に。

- L10N ガイドライン 
  ガイドライン冒頭に書かれている通り、Mozilla 製品の日本語ローカライズのためのガイドラインです。主に gecko-l10n リポジトリーと Pontoon にある製品が該当します。
- 表記ガイドライン 
  ガイドライン冒頭に書かれている通り、製品と公式サイトに適用されます。 SUMO は公式のサポートサイトで、ナレッジベース記事に書かれたインターフェースやエラーメッセージ等は製品の表記に従わなければなりません。

Github や Pontoon はただのツールです。gecko-l10n のリポジトリーも、trac、Google Code などいくつかのシステムを渡り歩いてきた経緯があり、それぞれのサイトで関連があるのは自然なことです。

**Comment by kazushi3** [comment]
@hmatrjp
@marsf

ありがとうございます。
コミュニティのありように関しての部分は、当初変な議論の仕方を試みてしまいました。私から投げた問いで、お返事をいただいた側が申し上げるべきではないのですが、議論すべきことから離れてしまいそうなので置いておかせてください。有意義な内容をいただいたと思っておりますので、しかるべき場所で役立てさせてください。わざわざお答えさせた上にうまく繋げられない失礼をお許しください。

追記の部分について、

> そのロードマップページの最終更新（末尾にあります）が4ヶ月前であることは注意しておいた方がいい

>その後、実績の反映がされていないかも

プレスリリースからも、遅れていると思っています。Q4 リリースは厳しそうです。ただ、セキュリティ監査の終了や予定画像を発表していた UI の完成はあるので、当初は 2023 年の夏が目標だったこともあり、大穴でとりあえず現状のものを Thunderbird の名前で出さないとも限りないのとも思っています。

>K-9 コミュニティのほうから何か応答がありましたか？

旧ツールの登録者にメッセージ機能で移動する予定があるか聞いてみてはいるのですが、今のところ応答はありません。
Weblate の仕様で、登録のみしてまだ翻訳していない方は確認できないのですが、翻訳実績は私のみです。

>「日本語コミュニティーでやりたい」と書かれた意図

「人数を集めたい」ことと「Thunderbird の翻訳の一貫性を維持すべき」と考えたことが主です。
K-9 Mail の旧翻訳ツール時代でも未翻訳は数十程度残っていた印象でした。(今見るとそうでもないので、勘違いや私が見たタイミングが悪かった可能性もあり) また、表記揺れや真逆の意味になる誤訳などもありました。

すべての Mozilla 製品が完訳されている訳ではないことからも、日本語コミュニティと関われば翻訳に興味を持つ人が増えるとは言えませんが、増えたらいいなとは思っています。言語ファイルが新規のパーツごとのファイルと ui-legacy に分かれていることや関係者の発言などから最終的に全 UI の置き換わりがあると予想しており、確実に追従するにはたくさんの方の力が必要になると思います。

Thunderbird の名前でリリースされることや、PC 版 Thunderbird との同期を搭載予定なことから私は Mozilla 製品としての翻訳の一貫性を持たせるべきと考えました。

GitHub の mozilla-japan/translation リポジトリーについては考えていませんでしたが、相談できる場所としてあれば便利かもしれません。

>ソースコードが Thunderbird 配下になったので、これで移行が完了したとみてよいと思います。

thundernest -> thunderbird 配下に移行され、リポジトリー名も k-9 から thunderbird-android になりました。
確認なのですが、Mozilla 日本語ローカライズの一部と言える状態になったとみてよいのでしょうか？

**Reply by kazushi3** [reply]
リリースは 2024 年の Q1 に遅れるようです。

https://blog.thunderbird.net/2023/11/the-untold-history-of-thunderbird/

**Comment by marsf** [comment]
> 確認なのですが、Mozilla 日本語ローカライズの一部と言える状態になったとみてよいのでしょうか？

「移行」というのは、双方の合意や取り決めがあって成り立つものだと思いますが。
はじめに書いた通り、コミュニティの合意はありましたか？
相手の意向も分からないのに、それをどうやったら判断できますか？

Kazushi さんは、コミュニティの活動を「引き取る」という難しい選択をされています。
道筋が立てられていればこのような質問は出てこないはずです。
一度立ち止まって、目的のために何ができるのか考え直してみてはいかがでしょうか。

**Comment by kazushi3** [comment]
道筋としては、marsf さんから最初にいただいたアドバイスでの 2 点

>K-9 Mail は Mozilla とは別のプロジェクトでしたから、移行期間の今はまだ Mozilla 日本語ローカライズの一部と言えるかどうか微妙なところです。

>次に、K-9 Mail は Mozilla とは別のプロジェクトとして開発されてきましたから、Mozilla コミュニティとは別にユーザーのコミュニティがあって、これまでのローカライズに貢献してきた人がいます。

を元に

1, K-9 が Mozilla 製品である状態になり、Mozilla 日本語ローカライズの範囲内になる
2, 現 K-9 のユーザーコミュニティと話し合い、今後の取り扱いについて何らかの合意に達する

の 2 ステップが必要と考えています。

今回ご指摘を受けた私の質問は、 *ソースコードが Thunderbird 配下になったので、これで移行が完了したとみてよいと思います。* というのは *移行期間の今はまだ Mozilla 日本語ローカライズの一部と言えるかどうか微妙なところです。* と仰られたことに対応し、先の道筋での 1 は満たしたと判断してよいですか、という趣旨でした。書き方が悪く申し訳ありません。

コミュニティの合意については、同じコメント内でお答えした *旧ツールの登録者にメッセージ機能で移動する予定があるか聞いてみてはいるのですが、今のところ応答はありません。* というのがすべてで合意には至っておらず、全タスクが完了したとは思っておりません。

**Comment by marsf** [comment]
> * 移行期間の今はまだ Mozilla 日本語ローカライズの一部と言えるかどうか微妙なところです。 
> * ソースコードが Thunderbird 配下になったので、これで移行が完了したとみてよいと思います。 

どちらも、「K-9 プロジェクト本体」が Thunderbird の開発下へ移行することを指しています。
これは、ソースコードが Thunderbird の開発下になり、作者からのメッセージがあって完了しました。

> We've successfuly completed the move of our source code repository to https://github.com/thunderbird/thunderbird-android. Translation have been unlocked,

「Mozilla 日本語ローカライズの一部」になるかどうかは、上記のプロジェクト本体の移行という前提条件なくしてはありえないことです。加えて、Kazushi さんが最初に作者に確認された通り、K-9 プロジェクト独自のローカライズの方針は何も変わっていないように見えます。

仮に作者から、(まず有り得ないことですが) 今後は Mozilla のローカライズコミュニティに作業してほしいという意向が示されて Mozilla 日本語ローカライズの一部になったとしても、ローカライズの主導権を現コミュニティからこちらに勝手に移すことは許可できません。
K-9 の現コミュニティとの合意なしでこのような身勝手が通れば、プロジェクトそのものが崩壊します。

> 1, K-9 が Mozilla 製品である状態になり、Mozilla 日本語ローカライズの範囲内になる

そういうわけで、まずステップ 1 が成立しません。
何度も書いているように、優先すべきなのはステップ 2 だと思います。

**Comment by kazushi3** [comment]
ありがとうございます。

「Mozilla 日本語ローカライズの一部」というところの意味を遅ればせながら理解できました。

現コミュニティは、まだリアクションがないため言及が少ないだけで軽視してはないつもりです。
実際に翻訳に関わっていた方の意見を聞きたいと考えたため、個別のメッセージ機能とメンテナーさんへのメール送信を行っています。
翻訳には関わっていない方もいらっしゃる Google グループへ投稿すると混乱するかもしれないと思いまだ保留しているのですが、このまま反応がなければ行おうと思います。
